projectq/ops/_gates.py

```diff
@@ -74,17 +77,29 @@ def matrix(self):
 
 class XGate(SelfInverseGate):
     """ Pauli-X gate class """
+
     def __str__(self):
         return "X"
 
     @property
     def matrix(self):
         return np.matrix([[0, 1], [1, 0]])
 
+    def get_commutable_circuit_list(self, n=0):
+        """ Sets _commutable_circuit_list for C(NOT, n) where
+        n is the number of controls """
+        if (n == 1):
+            # i.e. this is a CNOT gate (one control)
+            # We define the qubit with the NOT as qb0, the qubit with 
+            # the control as qb1, then all next qbs are labelled 2,3 etc.
+            commutable_circuit_list = [[RelativeCommand(H,(0,)), RelativeCommand(C(NOT),(2,), relative_ctrl_idcs=(0,)), RelativeCommand(H,(0,))]]
+            return commutable_circuit_list
+        else:
+            return [] # don't change _commutable_circuit_list
+
 #: Shortcut (instance of) :class:`projectq.ops.XGate`
 X = NOT = XGate()
 
-
 class YGate(SelfInverseGate):
```

Comment:
Missing commutation with `Y`, `Ry` and `Ryy`

Reply:
Done. As will be seen in next commit. I haven't listed Y as a gate which commutes with YGate (or put in logic for any gates to commute with themselves) because I can't think of any situations where that would be useful.